frontend/javascripts/oxalis/geometries/materials/node_shader.ts

```diff
@@ -157,6 +185,13 @@ vec3 shiftHue(vec3 color, float shiftValue) {
 }
 
 void main() {
+    <% _.each(additionalCoordinates || [], (_coord, idx) => { %>
+      if (additionalCoord_<%= idx %> != currentAdditionalCoord_<%= idx %>) {
+        return;
```

Comment:
I introduced the alpha blending so that edges that go from t=1 to t=2 are only half-visible (i.e., there's a linear gradient from 0% to 100%). Using the same alpha mechanism for invisible trees would very likely work, but I wouldn't want to risk potential performance hits. For example, when having a huge annotation where most skeletons are invisible, I imagine less effort the fragment shader when vertices are outside of the viewport (in constrast to being scattered everywhere with alpha=0).
I might be wrong about this, but I don't think challenging the above assumption is really worth it :see_no_evil:

Reply:
Ok, thanks for explaining, the "half-visible" part was not apparent to me. I would also assume that the shader won't become _more_ performant using alpha instead of moving the vertices outside of the viewport.